x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/KqlAstBuilder.java

```diff
@@ -8,22 +8,222 @@
+        String queryText = ParserUtils.extractText(ctx.fieldQueryValue());
+
+        if (hasWildcard(ctx.fieldQueryValue())) {
+            // TODO: set default fields.
+            return QueryBuilders.queryStringQuery(escapeLuceneQueryString(queryText, true));
+        }
+
+        boolean isPhraseMatch = ctx.fieldQueryValue().QUOTED_STRING() != null;
+
+        return QueryBuilders.multiMatchQuery(queryText)
+            // TODO: add default fields?
+            .type(isPhraseMatch ? MultiMatchQueryBuilder.Type.PHRASE : MultiMatchQueryBuilder.Type.BEST_FIELDS)
+            .lenient(true);
+    }
+
+    @Override
+    public QueryBuilder visitFieldQuery(KqlBaseParser.FieldQueryContext ctx) {
+
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery().minimumShouldMatch(1);
+        String queryText = ParserUtils.extractText(ctx.fieldQueryValue());
+        boolean hasWildcard = hasWildcard(ctx.fieldQueryValue());
+
+        withFields(ctx.fieldName(), (fieldName, mappedFieldType) -> {
+            QueryBuilder fieldQuery = null;
+
+            if (hasWildcard && isKeywordField(mappedFieldType)) {
+                fieldQuery = QueryBuilders.wildcardQuery(fieldName, queryText)
+                    .caseInsensitive(kqlParserExecutionContext.isCaseSensitive() == false);
+            } else if (hasWildcard) {
+                fieldQuery = QueryBuilders.queryStringQuery(escapeLuceneQueryString(queryText, true)).field(fieldName);
+            } else if (isDateField(mappedFieldType)) {
+                // TODO: add timezone
+                fieldQuery = QueryBuilders.rangeQuery(fieldName).gte(queryText).lte(queryText);
+            } else if (isKeywordField(mappedFieldType)) {
+                fieldQuery = QueryBuilders.termQuery(fieldName, queryText)
+                    .caseInsensitive(kqlParserExecutionContext.isCaseSensitive() == false);
+            } else if (ctx.fieldQueryValue().QUOTED_STRING() != null) {
+                fieldQuery = QueryBuilders.matchPhraseQuery(fieldName, queryText);
+            } else {
+                fieldQuery = QueryBuilders.matchQuery(fieldName, queryText);
+            }
+
+            if (fieldQuery != null) {
+                boolQueryBuilder.should(fieldQuery);
+            }
+        });
+
+        return rewriteDisjunctionQuery(boolQueryBuilder);
+    }
+
+    private static boolean isAndQuery(KqlBaseParser.BooleanQueryContext ctx) {
+        return ctx.operator.getType() == KqlBaseParser.AND;
+    }
+
+    private static boolean isOrQuery(KqlBaseParser.BooleanQueryContext ctx) {
+        return ctx.operator.getType() == KqlBaseParser.OR;
+    }
+
+    private void withFields(KqlBaseParser.FieldNameContext ctx, BiConsumer<String, MappedFieldType> fieldConsummer) {
+        kqlParserExecutionContext.resolveFields(ctx).forEach(fieldDef -> fieldConsummer.accept(fieldDef.v1(), fieldDef.v2()));
+    }
+
+    private QueryBuilder rewriteDisjunctionQuery(BoolQueryBuilder boolQueryBuilder) {
```

Comment:
Testing is a lot more simpler when having the query rewritten IMHO,

We can write things like:

```
MatchQueryBuilder parsedQuery = asInstanceOf(MatchQueryBuilder.class, parseKqlQuery("foo: bar"));
```

instead of having to unfold the should clauses of the query:

```
MatchQueryBuilder parsedQuery = asInstanceOf(
  MatchQueryBuilder.class, 
  asInstanceOf(BoolQueryBuilder.class, parseKqlQuery("foo: bar")).should().getFirst()
);
```